# 1.3 Sparing

## Eksempel x1

Boligsparingsprosjekt i BSU

Rente: 3.10%
Årlig innskudd 27 500 kr

Varighet: 8 år


*Hva er saldoen i slutten av det 8. året?*


Saldoen etter 8 år med innskud og renter er 253016.40 kr

*Hva er saldoen rett etter det 89. innskuddet?*

In [6]:
saldo = 0
vf = 1.031 
innskudd = 27500

print("Saldo rett etter hvert innskudd:")

for år in range(1, 9):
    saldo = saldo + innskudd
    print(f"{år}: {saldo:.2f}")
    saldo = saldo*vf
    

Saldo rett etter hvert innskudd:
1: 27500.00
2: 55852.50
3: 85083.93
4: 115221.53
5: 146293.40
6: 178328.49
7: 211356.68
8: 245408.73


## Eksempel 11

In [15]:
saldo = 0
innskudd = 20000
vf = 1.04

print("Saldo i slutten av hvert år:")

for år in range(2022, 2036):
    saldo = saldo + innskudd
    saldo = saldo*vf

    #print(år, ":", round(saldo, 2), "kr")
    print(f"{år}: {saldo:9.2f} kr")


Saldo i slutten av hvert år:
2022:  20800.00 kr
2023:  42432.00 kr
2024:  64929.28 kr
2025:  88326.45 kr
2026: 112659.51 kr
2027: 137965.89 kr
2028: 164284.53 kr
2029: 191655.91 kr
2030: 220122.14 kr
2031: 249727.03 kr
2032: 280516.11 kr
2033: 312536.75 kr
2034: 345838.22 kr
2035: 380471.75 kr


### Oppgave 1.21

In [23]:
saldo = 0
innskudd = 3000
vf = 1.05

print("Saldo rett etter hvert innskudd:")

for år in range(1, 11):
    saldo = saldo + innskudd
    print(f"{år:2}: {saldo:7.2f} kr")
    saldo = saldo*vf


Saldo rett etter hvert innskudd:
 1: 3000.00 kr
 2: 6150.00 kr
 3: 9457.50 kr
 4: 12930.38 kr
 5: 16576.89 kr
 6: 20405.74 kr
 7: 24426.03 kr
 8: 28647.33 kr
 9: 33079.69 kr
10: 37733.68 kr


Saldoen rett etter det 10. innskuddet er 37733.68 kr.

## Oppgave 1.23

In [24]:
#a
utslipp = 58 # i tonn
vf = 0.94

for år in range(1, 11):
    utslipp = utslipp*vf
    print(f"{år:2}: {utslipp:.1f}")

 1: 54.5
 2: 51.2
 3: 48.2
 4: 45.3
 5: 42.6
 6: 40.0
 7: 37.6
 8: 35.4
 9: 33.2
10: 31.2


Bedriften slipper ut 42.6 tonn CO2 det 5. året.

In [27]:
#b

utslipp = 58 # i tonn
vf = 0.94
sum = 0

for år in range(1, 11):
    utslipp = utslipp*vf
    sum = sum + utslipp
    #print(f"{år:2}: {utslipp:.1f}")

print(f"Summen av utslippene de 10 neste åprene: {sum:.1f} tonn.")

Summen av utslippene de 10 neste åprene: 419.2 tonn.


## Oppgave 1.31

In [1]:
lån = 4500000
terminbeløp = 280000
vf = 1.03

for år in range(1, 24):
    lån = lån*vf - terminbeløp
    print(f"{år:2}: {lån:11.2f} kr")

 1:  4355000.00 kr
 2:  4205650.00 kr
 3:  4051819.50 kr
 4:  3893374.08 kr
 5:  3730175.31 kr
 6:  3562080.57 kr
 7:  3388942.98 kr
 8:  3210611.27 kr
 9:  3026929.61 kr
10:  2837737.50 kr
11:  2642869.62 kr
12:  2442155.71 kr
13:  2235420.38 kr
14:  2022483.00 kr
15:  1803157.49 kr
16:  1577252.21 kr
17:  1344569.78 kr
18:  1104906.87 kr
19:   858054.08 kr
20:   603795.70 kr
21:   341909.57 kr
22:    72166.86 kr
23:  -205668.14 kr


In [38]:
lån = 4500000
terminbeløp = 40000
pf = 0.03
år = 1

while lån > 0:
    avdrag = terminbeløp - lån*pf
    lån = lån - avdrag
    print(f"{år:2}: {lån:11.2f}")
    år = år + 1


 1:  4355000.00
 2:  4205650.00
 3:  4051819.50
 4:  3893374.08
 5:  3730175.31
 6:  3562080.57
 7:  3388942.98
 8:  3210611.27
 9:  3026929.61
10:  2837737.50
11:  2642869.62
12:  2442155.71
13:  2235420.38
14:  2022483.00
15:  1803157.49
16:  1577252.21
17:  1344569.78
18:  1104906.87
19:   858054.08
20:   603795.70
21:   341909.57
22:    72166.86
23:  -205668.14


In [45]:
lån = 9000000
terminbeløp = 39000 #månedlig
pf = 0.002871 #månedlig (3.5% årlig rente tilsvarer 0.2871% i måneden)
måned = 0

while lån > 0:
    måned = måned + 1
    avdrag = terminbeløp - lån*pf
    lån = lån - avdrag
    print(f"{måned:3}: {lån:11.2f} kr")

print(måned/12)

  1:  8986839.00 kr
  2:  8973640.21 kr
  3:  8960403.54 kr
  4:  8947128.85 kr
  5:  8933816.06 kr
  6:  8920465.05 kr
  7:  8907075.70 kr
  8:  8893647.92 kr
  9:  8880181.58 kr
 10:  8866676.58 kr
 11:  8853132.81 kr
 12:  8839550.15 kr
 13:  8825928.50 kr
 14:  8812267.74 kr
 15:  8798567.76 kr
 16:  8784828.45 kr
 17:  8771049.69 kr
 18:  8757231.38 kr
 19:  8743373.39 kr
 20:  8729475.61 kr
 21:  8715537.94 kr
 22:  8701560.25 kr
 23:  8687542.43 kr
 24:  8673484.36 kr
 25:  8659385.94 kr
 26:  8645247.03 kr
 27:  8631067.54 kr
 28:  8616847.33 kr
 29:  8602586.30 kr
 30:  8588284.33 kr
 31:  8573941.29 kr
 32:  8559557.08 kr
 33:  8545131.56 kr
 34:  8530664.64 kr
 35:  8516156.17 kr
 36:  8501606.06 kr
 37:  8487014.17 kr
 38:  8472380.39 kr
 39:  8457704.59 kr
 40:  8442986.66 kr
 41:  8428226.48 kr
 42:  8413423.91 kr
 43:  8398578.85 kr
 44:  8383691.17 kr
 45:  8368760.75 kr
 46:  8353787.46 kr
 47:  8338771.19 kr
 48:  8323711.80 kr
 49:  8308609.18 kr
 50:  8293463.19 kr


In [7]:
pf = 0.0155
lån = 20000
terminbeløp = 729

for mnd in range(1, 37):
    avddrag = terminbeløp - lån*pf
    lån = lån - avddrag
    print(f"{mnd:3}: {lån:10.2f} kr")

  1:   19581.00 kr
  2:   19155.51 kr
  3:   18723.42 kr
  4:   18284.63 kr
  5:   17839.04 kr
  6:   17386.55 kr
  7:   16927.04 kr
  8:   16460.41 kr
  9:   15986.54 kr
 10:   15505.33 kr
 11:   15016.67 kr
 12:   14520.42 kr
 13:   14016.49 kr
 14:   13504.75 kr
 15:   12985.07 kr
 16:   12457.34 kr
 17:   11921.43 kr
 18:   11377.21 kr
 19:   10824.56 kr
 20:   10263.34 kr
 21:    9693.42 kr
 22:    9114.67 kr
 23:    8526.94 kr
 24:    7930.11 kr
 25:    7324.03 kr
 26:    6708.55 kr
 27:    6083.53 kr
 28:    5448.83 kr
 29:    4804.29 kr
 30:    4149.75 kr
 31:    3485.07 kr
 32:    2810.09 kr
 33:    2124.65 kr
 34:    1428.58 kr
 35:     721.72 kr
 36:       3.91 kr


In [9]:
vf = 1.0155
lån = 20000 #x
terminbeløp = 729

for mnd in range(1, 37):
    lån = lån * vf - terminbeløp
    print(f"{mnd:3}: {lån:10.2f} kr")

  1:   19581.00 kr
  2:   19155.51 kr
  3:   18723.42 kr
  4:   18284.63 kr
  5:   17839.04 kr
  6:   17386.55 kr
  7:   16927.04 kr
  8:   16460.41 kr
  9:   15986.54 kr
 10:   15505.33 kr
 11:   15016.67 kr
 12:   14520.42 kr
 13:   14016.49 kr
 14:   13504.75 kr
 15:   12985.07 kr
 16:   12457.34 kr
 17:   11921.43 kr
 18:   11377.21 kr
 19:   10824.56 kr
 20:   10263.34 kr
 21:    9693.42 kr
 22:    9114.67 kr
 23:    8526.94 kr
 24:    7930.11 kr
 25:    7324.03 kr
 26:    6708.55 kr
 27:    6083.53 kr
 28:    5448.83 kr
 29:    4804.29 kr
 30:    4149.75 kr
 31:    3485.07 kr
 32:    2810.09 kr
 33:    2124.65 kr
 34:    1428.58 kr
 35:     721.72 kr
 36:       3.91 kr


In [6]:
saldo = 0
vf = 1.03
innskudd = 8000

for år in range(1,6):
    saldo = (saldo + innskudd)*vf
    print(f"{år}: {saldo:10.2f} kr")

1:    8240.00 kr
2:   16727.20 kr
3:   25469.02 kr
4:   34473.09 kr
5:   43747.28 kr


In [5]:
saldo = 0
vf = 1.03
innskudd = 8000
år = 0

while saldo < 60000:
    år = år + 1
    saldo = (saldo + innskudd)*vf
    print(f"{år}: {saldo:10.2f} kr")

1:    8240.00 kr
2:   16727.20 kr
3:   25469.02 kr
4:   34473.09 kr
5:   43747.28 kr
6:   53299.70 kr
7:   63138.69 kr


In [7]:
saldo = 0
for år in range(1,6):
    saldo = (saldo + 8000)*1.03
    
print(f"{år}: {saldo:10.2f} kr")

5:   43747.28 kr


In [4]:

saldo = 0
for år in range(1,18):
    saldo = (saldo + 20000)*1.015
    

#print(f"{år}: {saldo:10.2f} kr")

saldo = saldo + 20000 # Det siste innskuddet uten renter

print(f"Saldo rett etter det 18. innskuddet: {saldo:.2f} kr")

Saldo rett etter det 18. innskuddet: 409787.51 kr


In [7]:
saldo = 0
år = 0

while saldo < 250000:
     saldo = (saldo + 20000)*1.015
     år = år + 1


print(f"År: {år}    Saldo: {saldo:.2f}")

År: 12    Saldo: 264736.59


In [12]:
personer = [
["Per", "Spelmann", "48år", "Oslo"],
["Gunn", "Johansen", "66år", "Røros"],
["Ivar", "Aasen", "205år", "Åsen"],
["Olav", "Tryggvason", "97år", "Trondheim"],
["Nils", "Nilsen", "19år", "Bergen"]]

print(personer[-1])

['Nils', 'Nilsen', '19år', 'Bergen']


In [13]:
import numpy as np

arr = np.array([[1, 2, 3, 4], [5, 6, 7, 8]])

print(arr.shape)

(2, 4)


In [17]:
import numpy as np

liste = [["John", "Steinar"], ["Jon", "Arild"]]
arr = np.array(liste)

print(arr.shape)

(2, 2)
